src/shipping/src/telemetry_conf.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use log::Level;
+use opentelemetry::global;
+use opentelemetry_appender_log::OpenTelemetryLogBridge;
+
+use opentelemetry_resource_detectors::{OsResourceDetector, ProcessResourceDetector};
+use opentelemetry_sdk::{
+    propagation::TraceContextPropagator, resource::ResourceDetector, Resource,
+};
+
+use std::{env, str::FromStr};
+
+fn get_resource() -> Resource {
+    let detectors: Vec<Box<dyn ResourceDetector>> = vec![
+        Box::new(OsResourceDetector),
+        Box::new(ProcessResourceDetector),
+    ];
+
+    Resource::builder().with_detectors(&detectors).build()
+}
+
+fn init_tracer_provider() {
+    global::set_text_map_propagator(TraceContextPropagator::new());
+
+    let tracer_provider = opentelemetry_sdk::trace::SdkTracerProvider::builder()
+        .with_resource(get_resource())
+        .with_batch_exporter(
+            opentelemetry_otlp::SpanExporter::builder()
+                .with_tonic()
+                .build()
+                .expect("Failed to initialize tracing provider"),
+        )
+        .build();
+
+    global::set_tracer_provider(tracer_provider);
+}
+
+fn init_meter_provider() -> opentelemetry_sdk::metrics::SdkMeterProvider {
+    let meter_provider = opentelemetry_sdk::metrics::SdkMeterProvider::builder()
+        .with_resource(get_resource())
+        .with_periodic_exporter(
+            opentelemetry_otlp::MetricExporter::builder()
+                .with_temporality(opentelemetry_sdk::metrics::Temporality::Delta)
+                .with_tonic()
+                .build()
+                .expect("Failed to initialize metric exporter"),
+        )
+        .build();
+    global::set_meter_provider(meter_provider.clone());
+
+    meter_provider
+}
+
+fn init_logger_provider() {
+    let logger_provider = opentelemetry_sdk::logs::SdkLoggerProvider::builder()
+        .with_resource(get_resource())
+        .with_batch_exporter(
+            opentelemetry_otlp::LogExporter::builder()
+                .with_tonic()
+                .build()
+                .expect("Failed to initialize logger provider"),
+        )
+        .build();
+
+    let otel_log_appender = OpenTelemetryLogBridge::new(&logger_provider);
```

Comment:
consider using `tracing` for logs + tracing-appender here.

Reply:
> the main goal of this PR is removing the tracing dependency.

I guess you are removing the idea of creating Otel Spans using `tracing` crate.

For logging, either `log` or `tracing` crate can be used. There are appenders for both in the Otel repo. But most examples maintained in the Otel Rust repo are using `tracing` for logs, so I think its good if demo also uses same, though this is not a hard requirement at all. (Also Otel Rust itself is using `tracing` for its internal logs)